### PR TITLE
Add resources to instrumentation CR

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Notable changes:
* add resources to instrumentation CR
* automatically inject some k8s resources

### CODE REVIEW

**Comment by pavolloffay** [comment]
@anuraaga would you like to review this one?

**Comment by pavolloffay** [comment]
@VineethReddy02 it seems like there is a flaky test

```
=== RUN   Test0_36_0Upgrade
--- PASS: Test0_36_0Upgrade (0.00s)
=== RUN   Test0_38_0Upgrade
    v0_38_0_test.go:149: 
        	Error Trace:	v0_38_0_test.go:149
        	Error:      	Not equal: 
        	            	expected: "upgrade to v0.38.0 dropped the deprecated logging arguments i.e. [--log-profile --log-format --log-level] from otelcol custom resource otelcol.spec.args and adding them to otelcol.spec.config.service.telemetry.logs, if no logging parameters are configured already."
        	            	actual  : "upgrade to v0.38.0 dropped the deprecated logging arguments i.e. [--log-level --log-profile --log-format] from otelcol custom resource otelcol.spec.args and adding them to otelcol.spec.config.service.telemetry.logs, if no logging parameters are configured already."
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-upgrade to v0.38.0 dropped the deprecated logging arguments i.e. [--log-profile --log-format --log-level] from otelcol custom resource otelcol.spec.args and adding them to otelcol.spec.config.service.telemetry.logs, if no logging parameters are configured already.
        	            	+upgrade to v0.38.0 dropped the deprecated logging arguments i.e. [--log-level --log-profile --log-format] from otelcol custom resource otelcol.spec.args and adding them to otelcol.spec.config.service.telemetry.logs, if no logging parameters are configured already.
        	Test:       	Test0_38_0Upgrade
```

File: api/instrumentation/v1alpha1/instrumentation_types.go [hunk 1]
```diff
@@ -23,6 +23,11 @@ type InstrumentationSpec struct {
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
 	Exporter `json:"exporter,omitempty"`
 
+	// Java defines configuration for java auto-instrumentation.
```
Comment (anuraaga) [inline api/instrumentation/v1alpha1/instrumentation_types.go:26]: Update

File: pkg/instrumentation/sdk.go [hunk 2]
```diff
@@ -60,10 +65,78 @@ func injectCommonSDKConfig(otelinst v1alpha1.Instrumentation, pod corev1.Pod) co
...
 
+// createResourceMap creates resource attribute map.
+// User defined attributes (in explicitly set env var) have higher precedence.
+func createResourceMap(otelinst v1alpha1.Instrumentation, ns corev1.Namespace, pod corev1.Pod) map[string]string {
+	// get existing resources env var and parse it into a map
+	existingRes := map[string]bool{}
+	existingResourceEnvIdx := getIndexOfEnv(pod.Spec.Containers[0].Env, envOTELResourceAttrs)
+	if existingResourceEnvIdx > -1 {
+		split := strings.Split(pod.Spec.Containers[0].Env[existingResourceEnvIdx].Value, ",")
+		for _, v := range split {
```
Comment (anuraaga) [inline pkg/instrumentation/sdk.go:91]: We should probably trim strings.

File: pkg/instrumentation/sdk_test.go [hunk 3]
```diff
@@ -91,12 +111,20 @@ func TestSDKInjection(t *testing.T) {
 									Name:  "OTEL_EXPORTER_OTLP_ENDPOINT",
 									Value: "explicitly_set",
 								},
+								{
+									Name:  "OTEL_RESOURCE_ATTRIBUTES",
+									Value: "foo=bar,k8s.container.name=other",
```
Comment (anuraaga) [inline pkg/instrumentation/sdk_test.go:116]: Let's have a test case with a trailing comma

Comment (pavolloffay) [reply pkg/instrumentation/sdk_test.go]: added here

**Comment by pavolloffay** [comment]
@anuraaga this is ready for re-review

**Review by anuraaga — APPROVED** [review]

**Comment by pavolloffay** [comment]
@VineethReddy02 @jpkrohling could you please merge this one?

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**